### ISSUE TITLE
Cannot launch linux on x905x3 (TX3) 

### ISSUE THREAD

**Issue by Seahorde** [ophub/amlogic-s9xxx-armbian#1361]
**Device Information **
 - SOC: S095x3
 - Model TX3
 - OEM name: Bqeel Y8 PRO. The PCB is identical to **TANIX TX-3** with 4+32G
 - Stock Android OS starts and works fine
 - Front of the PCB: https://drive.google.com/file/d/1Ve7rryVoa1HutUACGeXz8SrycUrQUcDi/view?usp=share_link
 - Back of the PCB: https://drive.google.com/file/d/1NryxvxQH0rVS46qxjojBsy-2MhUL-7xT/view?usp=share_link

**Armbian Version **
 - Kernel Version: 5.15.109
 - Release: jammy

**Describe the bug **
Linux won't boot from SD card. Display blinks and the process starts over after 1-2 sec.

I downloaded Armbian_23.05.0_amlogic_s905x3_jammy_5.15.109_server_2023.04.27 and flashed it to SD.
Copy u-boot-tx3-bz.bin -> u-boot.ext
Edit uEnv.txt: FDT=/dtb/amlogic/meson-sm1-tx3-bz.dtb

Inserted SD into the TV box, pressed the reset button and rebooted.
I see blinking screen and the boot sequence starts over.
I have UART wires attached to the board. The serial traffic looks like this: https://drive.google.com/file/d/1HKfzkfeTG9A4QI-ShYpLLzeu0Redcbt0/view?usp=sharing

I tried other versions of Armbian, OpenWRT, LibreELEC with different combinations of boot loader images and dtb's. Every time I see the same behavior.

The only one system that worked was a CoreElec image:
CoreELEC-Amlogic-ng.arm-20.1-Nexus-Generic.img
copy device_trees/sm1_s905x3_2g.dtb -> /dtb.img
(Press the reset button, attach power, wait until CoreELEC logo shows up and release the reset button.)
The System loads, I see a working CoreElec on my screen and a Linux serial console is working too.
The serial console bootlog is here: https://drive.google.com/file/d/1zkz_pTpomyHjHgKFBvXg8JoPDdhy2omO/view?usp=sharing

I run compare tool on these boot logs and here's where they start diverging:
https://drive.google.com/file/d/1As7B_-8BRMFPg_fsAf7gh2OdPYMGk3Sn/view?usp=share_link

I think the issue is in the u-boot start scripts of all systems except CoreELEC.

Please, help me run Armbian on this system.
I saw some posts in forums that people successfully run Armbian on a TX3 and I just don't understand how it's possible.

PS:
I have one more Android TV. It's a S905W system with 2G/16G memory, PENDOO X8 mini with corrupted original Android. PCB is identical to X96 mini.
The behavior is exactly the same. CoreELEC image works fine but all other Linuxes won't.

**Comment by ophub** [comment]
armbian and openwrt can only be started in the Android system, if the CoreELEC system is installed in your box, it will not start. You need to flash back to the Android system first.

5.15.y kernel does not need to copy u-boot.ext


tx3 has Gigabit version and 100M version. But there is no need to modify it when starting. The default s905x3 can be started on tx3. You can choose the correct serial number when installing to emmc

**Comment by Seahorde** [comment]
I have not installed any systems on on eMMC. They're running stock Android. I am trying to start them via the SD card and hidden button trick.

Armbian, OpenWRT and LibreELEC SD card images won't start and go into infinite boot loop.
CoreELEC does start and work OK from an SD card.

Please, help me if you can. What am I missing?

**Comment by ophub** [comment]
In the document, see how to boot armbian from a U disk

**Comment by Seahorde** [comment]
I flashed Armbian_23.05.0_amlogic_s905x3_jammy_5.15.109_server_2023.04.27.img to a USB stick.
Specified meson-sm1-tx3-bz.dtb in uEnv.txt
Put the stick into a USB port of the TV box, pressed the hidden button and attached power.
The box is in the boot loop. Showing black screen and S905X3 logo repeatedly. Armbian won't start.
Tried another USB port - same thing. Same happens with that image on a microSD card.

This TANIX TX-3 box is starting a stock Android from eMMC just fine when no USB/SD connected.

Is there anything I'm doing wrong?
